azurerm/resource_arm_private_link_endpoint.go

```diff
@@ -169,18 +224,86 @@ func resourceArmPrivateLinkEndpointDelete(d *schema.ResourceData, meta interface
 		if response.WasNotFound(future.Response()) {
 			return nil
 		}
-		return fmt.Errorf("Error deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+		return fmt.Errorf("Error deleting Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
 		if !response.WasNotFound(future.Response()) {
-			return fmt.Errorf("Error waiting for deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+			return fmt.Errorf("Error waiting for deleting Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
 		}
 	}
 
 	return nil
 }
 
+func expandArmPrivateLinkEndpointServiceConnection(input []interface{}, parseManual bool) *[]network.PrivateLinkServiceConnection {
+	results := make([]network.PrivateLinkServiceConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateConnectonResourceId := v["private_connection_resource_id"].(string)
+		subresourceNames := v["subresource_names"].([]interface{})
+		requestMessage := v["request_message"].(string)
+		isManual := v["is_manual_connection"].(bool)
+		name := v["name"].(string)
+
+		if isManual == parseManual {
+			result := network.PrivateLinkServiceConnection{
+				Name: utils.String(name),
+				PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+					GroupIds:             utils.ExpandStringSlice(subresourceNames),
+					PrivateLinkServiceID: utils.String(privateConnectonResourceId),
+				},
+			}
+
+			if requestMessage != "" {
+				result.PrivateLinkServiceConnectionProperties.RequestMessage = utils.String(requestMessage)
+			}
+
+			results = append(results, result)
+		}
+	}
+
+	return &results
+}
+
+func flattenArmPrivateLinkEndpointServiceConnection(input *[]network.PrivateLinkServiceConnection) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		v := make(map[string]interface{})
+
+		if name := item.Name; name != nil {
+			v["name"] = *name
+		}
+		if props := item.PrivateLinkServiceConnectionProperties; props != nil {
+			if subresourceNames := props.GroupIds; subresourceNames != nil {
+				v["subresource_names"] = utils.FlattenStringSlice(subresourceNames)
+			}
+			if privateConnectionResourceId := props.PrivateLinkServiceID; privateConnectionResourceId != nil {
+				v["private_connection_resource_id"] = *privateConnectionResourceId
+			}
+			if requestMessage := props.RequestMessage; requestMessage != nil {
+				v["request_message"] = *requestMessage
+			}
+			if provisioningState := props.ProvisioningState; provisioningState != "" {
+				v["provisioning_state"] = provisioningState
+			}
```

Comment:
can we remove this field? Terraform should only be returning when this is fully provisioned

Reply:
While I was testing this I realized that I was pulling the wrong value back from azure this is actually `PrivateLinkServiceConnectionProperties.Description` which I have exposed as `request_response` to be consistent with the portal experience 

* `request_response` - Possible values are as follows:
   * `Auto-Approved` - The remote resource owner has added you to the `Auto-Approved` RBAC permission list for the remote resouce, all private link endpoint connection requests will be automatically `Approved`.

   * `Deleted state` - The resource owner has `Rejected` the private link endpoint connection request and has removed your private link endpoint request from the remote resource.

   * `request/response message` - If you submitted a manual private link endpoint connection request, while in the `Pending` status the `request_response` will display the same text from your `request_message` in the `private_service_connection` block above. If the private link endpoint connection request was `Rejected` by the owner of the remote resource, the text for the rejection will be displayed as the `request_response` text, if If the private link endpoint connection request was `Approved` by the owner of the remote resource, the text for the approval will be displayed as the `request_response` text